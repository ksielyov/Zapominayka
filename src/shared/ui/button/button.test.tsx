import React from 'react';
import renderer from 'react-test-renderer';
import {Button} from './index';
import {fireEvent, render} from '@testing-library/react-native';

it('button should render correctly', () => {
  const tree = renderer
    .create(
      <Button
        title={'Button test'}
        onPress={() => console.log('Button was pressed')}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('should call onPress function', () => {
  const buttonTitle = 'Button test';
  const onPressEvent = jest.fn();

  const {getByText} = render(
    <Button title={buttonTitle} onPress={onPressEvent} />,
  );

  fireEvent.press(getByText(buttonTitle));
  expect(onPressEvent).toHaveBeenCalled();
});
