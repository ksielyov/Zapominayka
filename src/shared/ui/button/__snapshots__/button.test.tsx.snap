// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderRadius": 196.17647058823528,
      "elevation": 7,
      "height": 137.3235294117647,
      "justifyContent": "center",
      "opacity": 1,
      "shadowColor": "rgba(0, 0, 0, 0.3)",
      "shadowOffset": Object {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 7,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000",
        "fontFamily": "Roboto-Regular",
        "fontSize": 31.388235294117646,
        "fontWeight": "400",
      }
    }
  >
    Button test
  </Text>
</View>
`;
